mscore/macos/cocoabridge.mm

```diff
@@ -20,14 +20,36 @@
 #include "cocoabridge.h"
 #import <Cocoa/Cocoa.h>
 
-void  CocoaBridge::setAllowsAutomaticWindowTabbing(bool flag)
-{
-    if ([NSWindow respondsToSelector:@selector(allowsAutomaticWindowTabbing)])
-        [NSWindow setAllowsAutomaticWindowTabbing: flag];
-}
+id<NSObject> observerToken;
 
-bool  CocoaBridge::isSystemDarkTheme()
-{
-    NSString *osxMode = [[NSUserDefaults standardUserDefaults] stringForKey:@"AppleInterfaceStyle"];
-    return ([osxMode isEqualToString:@"Dark"]);
-}
+void CocoaBridge::observeDarkModeSwitches(std::function<void()> f)
+      {
+      if (@available(macOS 10.14, *))
+            observerToken = [[NSDistributedNotificationCenter defaultCenter]
+                             addObserverForName:@"AppleInterfaceThemeChangedNotification" object:nil
+                             queue:nil usingBlock:^(NSNotification * _Nonnull note) {   f();   }];
+      }
+
+CocoaBridge::~CocoaBridge()
+      {
+      if (@available(macOS 10.14, *))
+            [[NSDistributedNotificationCenter defaultCenter] removeObserver:observerToken];
+      }
```

Comment:
This destructor is never called, because the `CocoaBridge` class is never instantiated. For that matter, we do not even need the constructor. This function should probably be renamed to something like `CocoaBridge::removeObservers()`, and called manually during the destruction of the `MuseScore` class. Sure, there is only one observer right now, but it is possible that we may want to add more in the future for other notifications. If and when that happens, it will not do to have an object called `observerToken`. We would need either a list of observer tokens or a more descriptive name for each one. I guess it depends on how much control we need over which observers are active at any given time.

Reply:
Oops! You're right. First, I referred to this StackOverflow post: https://stackoverflow.com/a/1008289/11432719. But then I changed my mind and decided not to use any instance of the CocoaBridge, but I forgot to do something with the destructor. Indeed, this way it doesn't make any sense. I will change this.

Reply:
Should be better now!